packages/flutter/lib/src/cupertino/nav_bar.dart

```diff
@@ -570,20 +662,32 @@ class _CupertinoNavigationBarState extends State<CupertinoNavigationBar> {
 
   @override
   Widget build(BuildContext context) {
-    final Color backgroundColor =
-      CupertinoDynamicColor.maybeResolve(widget.backgroundColor, context) ?? CupertinoTheme.of(context).barBackgroundColor;
+    // The static navigation bar does not expand or collapse (see CupertinoSliverNavigationBar),
+    // it will either display the collapsed nav bar with middle, or the expanded with largeTitle.
+    assert(widget.middle == null || widget.largeTitle == null);
```

Comment:
What if someone actually wanted a small middle text and a large title text at the same time (for whatever reason)?  Similar to `alwaysShowMiddle` in the sliver version.

Reply:
Given that the default in Apple is not to show both, I decided not to expose it at this time. If someone files an issue requesting it with a use case, we can add it later. This is definitely [lazy programming](https://github.com/flutter/flutter/blob/master/docs/contributing/Style-guide-for-Flutter-repo.md#lazy-programming). :)